### PR TITLE
Use config file instead of command line arguments

### LINKED ISSUE #4: Don't show password in process list
This had not been a concern for us as we're running this on a restricted access server already, but it's a common security issue nonetheless, so let's fix it.

### PR DESCRIPTION
**Pull request by bitfehler**
Use a temporary file with 0600 permissions to pass the credentials to
FreeIPMI instead of using the command line, which certainly constitutes
bad security practice.

Template the `driver-type` while at it to potentially support local IPMI
at some point.

Fixes #4

### CODE REVIEW

File: collector.go [hunk 1]
```diff
@@ -153,14 +155,44 @@ var (
 	)
 )
 
+func freeipmiConfig(driver, user, password string) string {
+	return fmt.Sprintf(`
+driver-type %s
+privilege-level admin
+username %s
+password %s
+	`, driver, user, password)
```
Comment (grobie) [inline collector.go]: Is the trailing tab in the config file on purpose?

File: collector.go [hunk 2]
```diff
@@ -153,14 +155,44 @@ var (
...
+}
+
+func freeipmiTempConfigFile(driver, user, password string) (string, error) {
+	content := []byte(freeipmiConfig(driver, user, password))
+	tmpfile, err := ioutil.TempFile("", "ipmi-exporter")
+	if err != nil {
+		return "", err
+	}
+	if _, err := tmpfile.Write(content); err != nil {
+		defer os.Remove(tmpfile.Name())
```
Comment (grobie) [inline collector.go]: Adding a new deferred function right before exiting appears to be unnecessary. Just call os.Remove() directory?

File: collector.go [hunk 3]
```diff
@@ -153,14 +155,44 @@ var (
...
+		return "", err
+	}
+	if err := tmpfile.Close(); err != nil {
+		defer os.Remove(tmpfile.Name())
```
Comment (grobie) [inline collector.go]: Same as above.

File: collector.go (hunk 1 above)
Comment (bitfehler) [reply collector.go]: Yes. There is a leading newline as well, i preferred the code readability.

File: collector.go (hunk 2 above)
Comment (bitfehler) [reply collector.go]: Good point

**Comment by bitfehler** [comment]
FreeIPMI does not support any other mechanism for this, AFAICT

**Comment by bitfehler** [comment]
Ah, _named_ pipe! That might work, checking.

**Pull request merged**